### PR TITLE
Add IN and NIN support

### PR DESCRIPTION
**Pull request by Ashtonian**
First attempt https://github.com/a8m/rql/pull/10/

The equality fn for the test doesn't like comparing slices because its deep, not sure what we want to do there. 

Looking at integration testing next.

### CODE REVIEW

**Comment by Ashtonian** [comment]
resolves https://github.com/a8m/rql/issues/8

**Comment by Ashtonian** [comment]
due to how args are handled by sql.query, this needs to be changed a bit. 

Currently assuming the parser receives valid rql it produces something like `Select * FROM table WHERE id > ? and name IN (?)`, and args would contain a slice as an argument `[5, [1,2]]`. `.query` wants something more like `Select * FROM table WHERE id > ? and name IN (?, ?)` and args to be flat `[5,1,2]`. 

related: https://stackoverflow.com/questions/20271123/how-to-execute-an-in-lookup-in-sql-using-golang/32837541

File: rql_test.go [hunk 1]
```diff
@@ -898,7 +923,7 @@ func equalExp(e1, e2 string) bool {
 
 func split(e string) []string {
 	var s []string
-	for len(e) > 0 {
+	for len(e) > 0 { // TODO: stuck in loop when parsing (?,?,?) instead of ?
```
Comment (Ashtonian) [inline rql_test.go]: Currently breaks

File: rql.go [hunk 2]
```diff
@@ -424,9 +424,24 @@ func (p *parseState) field(f *field, v interface{}) {
...
-		p.WriteString(p.fmtOp(f.Name, Op(opName[1:])))
-		p.values = append(p.values, f.CovertFn(opVal))
+
+		validateFn := f.ValidateFn
+		if opName == p.op(NIN) || opName == p.op(IN) {
+			validateFn = validateSlice(validateFn)
+		}
+		must(validateFn(opVal), "invalid datatype or format for field %q", f.Name)
+
+		if opName == p.op(NIN) || opName == p.op(IN) {
```
Comment (Ashtonian) [inline rql.go]: is there a better way to do this ?

**Comment by Ashtonian** [comment]
updated to support multiple `?` placeholders, added working integration tests, currently could use some clean up and the tests are broken, not sure why.

**Comment by jonathanroze** [comment]
Any news about this feature ? Thanks

**Comment by c84c** [comment]
> updated to support multiple `?` placeholders, added working integration tests, currently could use some clean up and the tests are broken, not sure why.

If you give me access to your fork, I can help you

**Comment by Ashtonian** [comment]
I'll take another stab at this pending review for https://github.com/a8m/rql/pull/49

**Pull request closed without merging**